core/trino-main/src/main/java/io/trino/execution/buffer/ArbitraryOutputBuffer.java

```diff
@@ -77,21 +73,21 @@
     //  The index of the first client buffer that should be polled
     private final AtomicInteger nextClientBufferIndex = new AtomicInteger(0);
 
-    private final StateMachine<BufferState> state;
+    private final OutputBufferStateMachine stateMachine;
```

Comment:
Commit message:
```
Add OutputBufferStateMachine
Encapsulate state transition logic for output buffer there
```
seemst to  be unfinished (or something is messed up :) )

Reply:
Updated